src/vtt-segment-loader.js

```diff
@@ -496,4 +500,30 @@ export default class VTTSegmentLoader extends SegmentLoader {
       };
     }
   }
+
+  handleRollover_(value, reference) {
```

Comment:
I wonder if it's worth making the rollover mechanic more generic and exposing it from mux, since we have almost the same thing [here](https://github.com/videojs/mux.js/blob/c10bb8cc484fd79340112ed86831a824f6e4c93c/lib/m2ts/timestamp-rollover-stream.js#L22)?

Reply:
On top of this, I feel like some documentation on what this function is doing, as well as the reasoning behind it, could be beneficial.

Reply:
@adrums86 

Yes, I was considering re-using mux.js, but I realized it should not be a public API from mux.js.

Ideally, it should be moved to video-utils and exported in both VHS and mux.js
Then I realized there would be no other places besides mux.js and VTT loader, and it would not be changed, so I added it here.

Do you think we could benefit from moving it to videojs-utils?

@wseymour15 this is basically 100% the same functionality we have in Mux.js in RolloverStream. But I can def add some details here as well.

Reply:
Yeah I mean, I think in an ideal world we would move this out into video-utils. In this specific case due to multiple factors I'm not sure it's worth the refactoring across multiple repositories. I'd say we leave it as is.

Reply:
@wseymour15 

> I feel like some documentation on what this function is doing, as well as the reasoning behind it, could be beneficial.

I added documentation